src/config/ConfigLoader.hh

```diff
@@ -0,0 +1,72 @@
+// Used to parse .ini configuration files into c++ types
+// Supports all basic types & strings and stl (non-associative) containers except forward_list
+// if used on a container existing elements are not removed
+// example use:
+
+// filename: example.ini
+//
+// myint = 5
+// mystr = hello
+
+// filename: client.cpp
+//
+// #include <ini.hpp>
+// #include <ConfigLoader.hh>
+// #include <string>
+// #include <iostream>
+//
+// ConfigLoader::Open("example.ini");
+// ConfigLoader::Load("myint", myint);
+// ConfigLoader::Load("myint", mystr);
+// ConfigLoader::Close();
+// std::cout << myint == 5 << std::endl; // 1
+// std::cout << mystr == "hello" << std::endl; // 1
+
+#ifndef __OXSX__CONFIG_LOADER__
+#define __OXSX__CONFIG_LOADER__
+#include <TypeTraits.hpp>
+#include <ini.hpp>
+#include <vector>
+#include <string>
+
+using TypeTraits::enable_if;
+using TypeTraits::is_container;
+using TypeTraits::is_number;
+
+class ConfigLoader{
+public:
+    void Open(const std::string& fileName_);
+    void Close();
+    
+    // Fundamental numeric types
+    template<typename TargetType>
+    static void
+    Load(const std::string& fieldName_,  TargetType& loadVal_, typename enable_if<is_number<TargetType>::value, int>::type = 0);
```

Comment:
Could upgrade this to anything with a `>>` operator? then it works for custom classes with `<<` defined